### PR TITLE
Feature/new header components

### PR DESCRIPTION
**Pull request by Clr34**

### CODE REVIEW

File: src/views/AccountPage.vue [hunk 1]
```diff
@@ -102,8 +91,7 @@ export default defineComponent({
   data (): AccountPageData {
     return {
       operations: [],
-      name: '',
-      editingTitle: false
+      amountInOperation: 0
```
Comment (ermoca) [inline src/views/AccountPage.vue:94]: ? not sure what purpose it serves. Which operation are we talking about ?

File: src/components/AccountPageHeader.vue [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+export default defineComponent({
+  name: 'AccountPageHeader',
+  components: {},
+  watch: {},
+  props: {
+    accountId: {
+      type: String,
+      required: true
+    },
+    totAccount: {
```
[the remark this answers is not in the document]
Comment (Clr34) [reply src/components/AccountPageHeader.vue:44]: Indeed tot is for total, I want to differentiate these two variables but I suppose is not necessary to take the difference...

File: src/views/AccountPage.vue (hunk 1 above)
Comment (Clr34) [reply src/views/AccountPage.vue:94]: So... actually, it doesn't use... I will remove it

File: src/components/BudgetHeader.vue [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    },
+    money: {
+      type: Number,
+      required: true
+    }
+  },
+  data (): BudgetHeaderData {
+    return {
+      budgetMonth: this.$props.month,
+      totalAllocated: this.totAllocated,
```
[the remark this answers is not in the document]
Comment (Clr34) [reply src/components/BudgetHeader.vue:73]: So... Actually, these datas are never used, I will remove it. In fact, props are directly used in the code

File: src/components/BudgetHeader.vue [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import { defineComponent } from 'vue'
+import Time from '@/utils/Time'
+import Utils from '@/utils/Utils'
+
+interface BudgetHeaderData {
+    budgetMonth: number;
+    totalAllocated: number;
+    totalSpent: number;
+    totalAvailable: number;
+    moneyToDivide: number;
```
[the remark this answers is not in the document]
Comment (Clr34) [reply src/components/BudgetHeader.vue:41]: Indeed it's better than my name, I am really bad to choose a good name's variable...

File: src/components/BudgetCmpt.vue [hunk 5]
```diff
@@ -1,46 +1,16 @@
...
-              <td class="name"><div>{{ $t("TOTAL") }}</div></td>
-              <td class="allocated">{{ getEurosAmount(this.totalBudgetData.allocated) }}</td>
-              <td class="spent">{{ getEurosAmount(this.totalBudgetData.spent) }}</td>
-              <td class="available">{{ getEurosAmount(this.totalBudgetData.available) }}</td>
-            </tr>
-          </tbody>
-        </table>
-      </div>
+      <BudgetHeader :month="this.budgetMonth" :totAllocated="totalAllocated" :totSpent="totalSpent" :totAvailable="totalAvailable" :money="moneyToDivide"
+      @change-month="switchMonth" />
```
[the remark this answers is not in the document]
Comment (Clr34) [reply src/components/BudgetCmpt.vue:5]: ok, i will change it

File: src/components/AccountPageHeader.vue [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+            </span>
+            <span class="validation col">
+              <button class="btn fas fa-check" v-on:click="updateName()"/>
+              <button class="btn fas fa-times" v-on:click="this.cancelEditing()"/>
+            </span>
+          </div>
+          <div v-else class="editableNameAccount">
+            <a v-on:click="this.displayTitleEditing()">
+              <h1>
+                {{ this.account ? this.account.name : "" }} :
```
[the remark this answers is not in the document]
Comment (Clr34) [reply src/components/AccountPageHeader.vue:15]: I don't see which condition is tested twice in this part of code. line 12 : the div is display only if 'editditngTitle' is false and line 13 : the a change the value of 'editingTitle' if the user click on it. They don't have the same job... or i don't understand

File: src/components/AccountPageHeader.vue [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+          </div>
+    </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue'
+import AccountService from '@/services/AccountService'
+import { Account } from '@/model/model'
+
+interface AccountPageHeader {
```
[the remark this answers is not in the document]
Comment (Clr34) [reply src/components/AccountPageHeader.vue]: fix in commit 3acea1cf33dd71ed1dc21e8e41d4603448963dc3

**Pull request merged**